src/Public/New-FunctionFromType.ps1

```diff
@@ -0,0 +1,153 @@
+
+        if ($Template -eq "Default") {
+            $Template = Resolve-Path -Path "$script:PSModuleRoot\Private\template.txt"
+        }
+        elseif ($Template -eq "dbatools") {
+            $Template = Resolve-Path -Path "$script:PSModuleRoot\Private\template-dbatools.txt"
+        }
+        else {
+            if (-not (Test-Path -Path $Template)) {
+                throw "Can't find $Template"
+            }
+        }
+        if ($Path -and -not (Test-Path -Path $Path)) {
+            $null = New-Item -ItemType Directory -Path $Path
+        }
+    }
+    process {
+        if ($PSCmdlet.ParameterSetName -ne 'TypeName') {
+            $TypeName = $InputObject.GetType().Name
+        }
+
+        $properties = @(
+            ($TypeName -as [Type]).GetProperties() | Where-Object { $_.CanWrite }
+            ($TypeName -as [Type]).GetFields()     | Where-Object { -not $_.Attributes.HasFlag([System.Reflection.FieldAttributes]::InitOnly) }
+        )
+
+        if ($ExcludeProperty) {
+            $properties = $properties | Where-Object Name -notin $ExcludeProperty
+        }
+
+        foreach ($verbname in $Verb) {
+            $text = Get-Content -Path $Template
+            $shortname = ($TypeName -split "\.")[-1]
+            $name = "$verbname-$Prefix" + $shortname
+            $text = $text.Replace("--name--", $name)
+            $text = $text.Replace("--confirmimpact--", $ConfirmImpact)
+            $params = $help = $process = @()
+            $filename = "$Path\$name.ps1"
+
+            foreach ($Property in $Properties) {
+                $propertyname = $Property.Name
+                if ($propertyname -eq "Parent") {
+                    $propertyname = "InputObject"
+                }
+                if ($Property.PropertyType.Name -as [Type]) {
+                    $type = $Property.PropertyType.Name
+                }
+                else {
+                    $type = $Property.PropertyType.FullName
+                }
+                if ($type -eq "Boolean") {
+                    $type = "Switch"
+                }
+
+                if ($propertyname -ne "InputObject") {
+                    $params += "     [{0}]`${1}" -f $type, $propertyname
+                }
+                else {
+                    $params += "     [parameter(ValueFromPipeline)]`n    [{0}]`${1}" -f $type, $propertyname
+                }
+                $help += "    .PARAMETER $($propertyname)`n`n    `n"
+                $process += "`$object.$($propertyname) = `$$($propertyname)"
+            }
+
+            $text = $text.Replace("--params--", ($params -join ",`n"))
+            $text = $text.Replace("--help--", $help)
+            $text = $text.Replace("--process--", ($process -join "`n"))
+            $text = $text.Replace("--fullname--", $TypeName)
+
+            if ($Passthru) {
+                $text
+            }
+            else {
+                $text | Set-Content -Path $filename
+                Get-ChildItem $filename
```

Comment:
I'm not a fan of outputting two different types of objects depending on the situation. I know this approach mirrors a bit of what New-Item and similar do where it'll return the FileInfo object for the item it's just created. I don't really have a better solution to this though other than to not emit that output but I think it might be quite useful for people since it'll confirm the creation of those files and where.

Reply:
Thanks for your time and the review! Generally, I also follow the same rule of thumb, but Passthru is pretty common, appearing in 63 of the 5800 commands on my system. During the livestream, I piped the GCI output to Code and just did some saves from here.

Reply:
That's a fair point. I'd initially considered just outputting the function definition as a string and letting the user handle it but this way is definitely more user friendly, especially when generating multiple functions from a single type. 

I'm happy to keep this behaviour in.